src/App.svelte

```diff
@@ -31,8 +32,17 @@
     connectionDialogState,
   } from './script/stores/connectDialogStore';
   import { isLoading } from 'svelte-i18n';
+  import { fetchCachedBrowserInfo } from './utils/api';
+  let isRedirectToNextGenDialogOpen: boolean = false;
 
   onMount(() => {
+    // Before or after compatibility (Ask Lucy)
+    (async () => {
+      const { country } = await fetchCachedBrowserInfo($consent);
+      isRedirectToNextGenDialogOpen = country === 'GB';
+      isRedirectToNextGenDialogOpen = true;
+    })();
```

Comment:
The [svelte tutorial for onMount ](https://svelte.dev/tutorial/onmount) uses an async function in the onMount. However, [this issue](https://github.com/sveltejs/svelte/issues/4927) suggests that the destroy function returned from onMount is not called if onMount is passed an async function, so I have wrapped the fetching with an async function.

Reply:
I think this is fine, but we don't have a destroy function for this onMount (i.e., we're not returning anything) so it won't make any difference.

Reply:
I think the situation is:
- if you need to return a destroy function for cleanup then you need to have a nested async step, or else the return value is a Promise not a function
- but state updates after destroy are harmless in Svelte (hard to find a good reference, please check in practice there is no logged error, e.g. with a timeout in a simple demo component)
-  so we don't need the destroy function and the top-level function can be async

Based largely on this answer and the comments below from the Svelte author: https://stackoverflow.com/questions/62087073/svelte-3-async-onmount-or-a-valid-alternative

So I think a simple version is fine.

Reply:
> but state updates after destroy are harmless in Svelte (hard to find a good reference, please check in practice there is no logged error, e.g. with a timeout in a simple demo component)

I have tested it in [Svelte's REPL](https://svelte.dev/repl/ef1d83f150ab462f96e229882e030f7b?version=4.2.12), and it looks like there's no logged error.

> I think this is fine, but we don't have a destroy function for this onMount (i.e., we're not returning anything) so it won't make any difference. 

Yeah, I wasn't sure whether to play it safe in case we do have a destroy function in the future

> so we don't need the destroy function and the top-level function can be async ... So I think a simple version is fine.

I reverted to using an async function in the onMount here: https://github.com/microbit-foundation/ml-trainer/pull/243/commits/287d2a33de1492e15d6837181f8e1a40e16216e0